v2/client.go

```diff
@@ -1183,6 +1186,49 @@ func (c *client) FetchMetadata(ctx context.Context) (*Metadata, error) {
 	return c.metadata.update(ctx)
 }
 
+// FetchPartitionMetadata returns the metadata of the partition. This is sent to
+// the partition's leader
+func (c *client) FetchPartitionMetadata(ctx context.Context, stream string, partition int32) (*PartitionInfo, error) {
+	var (
+		req           = &proto.FetchPartitionMetadataRequest{Stream: stream, Partition: partition}
+		partitionInfo = &PartitionInfo{}
+	)
+
+	err := c.doResilientLeaderRPC(ctx, func(client proto.APIClient) error {
+		resp, err := client.FetchPartitionMetadata(ctx, req)
+		if err != nil {
+			return err
+		}
+		// Get broker info from metadata
+		brokers := c.metadata.metadata.brokers
+
+		leader := resp.GetMetadata().GetLeader()
+		replicas := resp.GetMetadata().GetReplicas()
+		isr := resp.GetMetadata().GetIsr()
+
+		// complement replicas, isr with broker info
+
+		replicasInfo := make([]*BrokerInfo, 0, len(replicas))
+		isrInfo := make([]*BrokerInfo, 0, len(isr))
+		leaderInfo := brokers[leader]
```

Comment:
We should handle the edge case where `brokers[leader]` does not exist, otherwise we'll get a nil panic. Probably just return an error in that situation.

Reply:
Agree. I implemented the same thing when populating `isr` and `replicas`.